tests/framework/cli/pipeline/test_pipeline_pull.py

```diff
@@ -175,6 +195,64 @@ def test_pull_local_sdist_compare(
         assert not filecmp.dircmp(test_path, test_dest).diff_files
         assert source_params_config.read_bytes() == dest_params_config.read_bytes()
 
+    def test_pipeline_pull_same_alias_package_name(
+        self,
+        fake_project_cli,
+        fake_repo_path,
+        fake_metadata,
+    ):
+        call_pipeline_create(fake_project_cli, fake_metadata)
+        call_pipeline_package(fake_project_cli, fake_metadata)
+
+        sdist_file = (
+            fake_repo_path / "dist" / _get_sdist_name(name=PIPELINE_NAME, version="0.1")
+        )
+
+        result = CliRunner().invoke(
+            fake_project_cli,
+            [
+                "pipeline",
+                "pull",
+                str(sdist_file),
+                "--destination",
+                "tools",
+                "--alias",
+                PIPELINE_NAME,
+            ],
+            obj=fake_metadata,
+        )
+        assert result.exit_code == 0, result.stderr
+        assert "pulled and unpacked" in result.output
```

Comment:
We need to add more checks to these tests, to make sure the expected directory is created and the expected contained files exist. If you look at examples above we do `assert_package_files_exist` etc.